### ISSUE TITLE
VB CodeGenWithEvents.TestNameOfWithEventsSetterParameter is flakey

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#6214]
From master_win_dbg_unit64

```
System.IO.FileLoadException : API restriction: The assembly '' has already loaded from a different location. It cannot be loaded from a new location within the same appdomain.
+++++++++++++++++++
STACK TRACE:
at System.Reflection.RuntimeAssembly.nLoadImage(Byte[] rawAssembly, Byte[] rawSymbolStore, Evidence evidence, StackCrawlMark& stackMark, Boolean fIntrospection, SecurityContextSource securityContextSource) at System.Reflection.Assembly.ReflectionOnlyLoad(Byte[] rawAssembly) at Microsoft.CodeAnalysis.Test.Utilities.RuntimeAssemblyManager.LoadAsAssembly(ImmutableArray`1 rawAssembly, Boolean reflectionOnly) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\HostedRuntimeEnvironment.cs:line 619 at Microsoft.CodeAnalysis.Test.Utilities.RuntimeAssemblyManager.<>c__DisplayClass22_0.<GetAssembly>b__0(Guid _) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\HostedRuntimeEnvironment.cs:line 637 at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory) at Microsoft.CodeAnalysis.Test.Utilities.RuntimeAssemblyManager.GetAssembly(String fullName, Boolean reflectionOnly) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\HostedRuntimeEnvironment.cs:line 637 at Microsoft.CodeAnalysis.Test.Utilities.RuntimeAssemblyManager.GetMemberSignaturesFromMetadata(String fullyQualifiedTypeName, String memberName) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\HostedRuntimeEnvironment.cs:line 664 at Microsoft.CodeAnalysis.Test.Utilities.HostedRuntimeEnvironment.GetMemberSignaturesFromMetadata(String fullyQualifiedTypeName, String memberName) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\HostedRuntimeEnvironment.cs:line 388 at Microsoft.CodeAnalysis.Test.Utilities.MetadataSignatureUnitTestHelper.VerifyMemberSignatureHelper(HostedRuntimeEnvironment appDomainHost, String fullyQualifiedTypeName, String memberName, String& expectedSignature, List`1& actualSignatures) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\MetadataSignatureUnitTestHelper.cs:line 86 at Microsoft.CodeAnalysis.Test.Utilities.MetadataSignatureUnitTestHelper.VerifyMemberSignatures(HostedRuntimeEnvironment appDomainHost, SignatureDescription[] expectedSignatures) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\MetadataSignatureUnitTestHelper.cs:line 40 at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompilationVerifier.Emit(String expectedOutput, IEnumerable`1 manifestResources, Boolean peVerify, SignatureDescription[] expectedSignatures) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\CommonTestBase.CompilationVerifier.cs:line 109 at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Action`1 assemblyValidator, Action`1 symbolValidator, Boolean verify) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\CommonTestBase.cs:line 204 at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Boolean verify) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Test\Utilities\Desktop\CommonTestBase.cs:line 129 at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Boolean verify) in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 77 at Microsoft.CodeAnalysis.VisualBasic.UnitTests.CodeGenWithEvents.TestNameOfWithEventsSetterParameter() in d:\j\workspace\roslyn-internal_master_win_dbg_unit64\Open\src\Compilers\VisualBasic\Test\Emit\CodeGen\CodeGenWithEvents.vb:line 1149
```

**Comment by amcasey** [comment]
```TestHandlesForWithEventsFromBaseFromADifferentAssembly``` is also affected.

**Comment by jaredpar** [comment]
This should be fixed by #6503.  If it reoccurs outside of stabilization please re-open this bug.
